recipes-support/provisioning/files/phytec-board-config.sh

```diff
@@ -0,0 +1,414 @@
+
+do_resetAWSCONFIG() {
+    do_clearAWSCONFIG
+    do_initAWSCONFIG
+    return 0
+}
+
+do_onboarding() {
+    awsrequest="https://devices.aws.esec-experts.com/ownership/token"
+    if [ $# -eq 1 ]; then
+        awsrequest="--request POST https://devices.aws.esec-experts.com/ownership/customer_account?customer_mail=$1"
+    else
+        do_contract
+        if [ "$ESECCONTRACT" = False ]; then
+            return 0
+        fi
+    fi
+    do_initAWSCONFIG
+    PRIVATEKEYURI="pkcs11:model=SLB9670;manufacturer=Infineon;token=iotdm;object=iotdm-keypair;type=private"
+    response=$(curl --cacert ${AWSCONFIG_CERTPATH}/rootcert.pem --engine pkcs11 --key-type ENG --key "${PRIVATEKEYURI};pin-value=${TPM_PIN}" --cert ${AWSCONFIG_CERTPATH}/devcert.pem ${awsrequest})
+    if [ $? -ne 0 ]; then
+       return 1
+    fi
+    if [ "$INTERACTIVE" = True ]; then
+        #error check
+        check=$(echo ${response} | jq .message)
+        if [ "$check" = "\"Forbidden\"" ]; then
+            return 2
+        fi
+        check=$(echo ${response} | jq .errorType)
+        if [ ! -z "$check" ] && [ ! "$check" = null ]; then
+            check=$(echo ${response} | jq .errorMessage)
+            whiptail --msgbox "\
+Error Message from IoT Device Manager
+$check" $WT_HEIGHT $WT_WIDTH
+            return 0
+        fi
+        if [ $# -eq 0 ]; then
+            # get token
+            set_onboarded tokendevice
+            certserial=$(echo ${response} | jq .thing_name)
+            token=$(echo ${response} | jq .token)
+            valid=$(echo ${response} | jq .valid_until)
+            whiptail --msgbox "\
+Welcome to the ESEC IoT Device Manager Platform
+The next steps:
+ 1) Login to your account on the ESEC IoT Device
+    Manager Platform https://iot.aws.esec-experts.com.
+    If you have no account,
+    then use Option 1) New Account and OnBoarding
+ 2) Add the
+    Token: ${token}
+    Valid until: ${valid}
+    for the device with
+    Certificate serial number ${certserial}
+    to your account:
+" $WT_HEIGHT $WT_WIDTH
+        else
+            # get user item
+            set_onboarded accountdevice
+            check=$(echo ${response} | jq .user_item.customer_name)
+            whiptail --msgbox "\
+Welcome ${check}
+to the ESEC IoT Device Manager Platform
+The next steps:
+ 1) Check your email account for a message
+    from welcome@esec-experts.com
+ 2) Verify your email
+ 3) Login to your IoT Device Manager account
+ 4) Check the state of your device in
+    your IoT Device Manager account" $WT_HEIGHT $WT_WIDTH
+        fi
+    else
+        if [ $# -eq 0 ]; then
+            set_onboarded tokendevice
+        else
+            set_onboarded accountdevice
+        fi
+        echo "response: ${response}"
+    fi
+    return 0
+}
+
+do_newaccount() {
+    do_contract
+    if [ "${ESECCONTRACT}" = False ]; then
+        return 0
+    fi
+    account=$(whiptail --inputbox "Your Email-Adress for the account:" $WT_HEIGHT $WT_WIDTH 3>&1 1>&2 2>&3)
```

Comment:
"Your e-mail address for the new account:"

Reply:
is changed